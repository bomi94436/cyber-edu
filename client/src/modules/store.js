import { createStore, applyMiddleware } from "redux";
import reduceReducers from "reduce-reducers";
import { composeWithDevTools } from "redux-devtools-extension";
import ReduxThunk from "redux-thunk";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";

import initState from "./initState";
import main from "./reducers/main";
import user from "./reducers/user";

const persistConfig = {
  key: "root",
  storage,
};

const rootReducer = reduceReducers(initState, main, user);
const reducer = persistReducer(persistConfig, rootReducer);

export default function configureStore() {
  const store = createStore(
    reducer,
    composeWithDevTools(applyMiddleware(ReduxThunk))
  );
  const persistor = persistStore(store);
  return { store, persistor };
}
