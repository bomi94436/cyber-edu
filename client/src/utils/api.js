import axios from "axios";
import { USER_SERVER } from "../components/Config";

export const login = (dataToSubmit) =>
  axios.post(`${USER_SERVER}/login`, dataToSubmit);

export const register = (dataToSubmit) =>
  axios.post(`${USER_SERVER}/register`, dataToSubmit);
