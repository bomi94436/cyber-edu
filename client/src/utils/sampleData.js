export const register = [
  { title: "학생", name: "student", value: "professor" },
  { title: "학번(직번)", name: "studentId", value: "201795076" },
  { title: "비밀번호", name: "password", value: "bomibomi" },
  { title: "이름", name: "name", value: "정보미" },
  { title: "이메일", name: "email", value: "boo0729611@gmail.com" },
  { title: "휴대폰 번호", name: "phone", value: "01012345678" },
];
