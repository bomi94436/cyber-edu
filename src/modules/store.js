import { createStore, applyMiddleware } from "redux";
import reduceReducers from "reduce-reducers";
import { composeWithDevTools } from "redux-devtools-extension";
import ReduxThunk from "redux-thunk";
import initState from "./initState";
import main from "./main";

const reducer = reduceReducers(initState, main);

export default createStore(
  reducer,
  composeWithDevTools(applyMiddleware(ReduxThunk))
);
