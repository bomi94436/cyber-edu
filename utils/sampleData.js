const getStudentId = () => {
  let studentId = "";
  for (let i of [...Array(9).keys()]) {
    studentId += String(Math.floor(Math.random() * 10));
  }
  return studentId;
};

exports.register = [
  { title: "학번(직번)", name: "studentId", value: getStudentId() },
  { title: "비밀번호", name: "password", value: "bomibomi9" },
  { title: "이름", name: "name", value: "정보미" },
  { title: "이메일", name: "email", value: "boo0729611@gmail.com" },
  { title: "휴대폰 번호", name: "phone", value: "010-1234-5678" },
  { title: "학생", name: "role", value: "student" },
];

exports.registerList = [
  getStudentId(),
  "bomibomi",
  "정보미",
  "boo0729611@gmail.com",
  "010-1234-5678",
  "professor",
];
