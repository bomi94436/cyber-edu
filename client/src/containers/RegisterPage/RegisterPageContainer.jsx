import RegisterPage from "../../components/views/RegisterPage/RegisterPage";
import { connect } from "react-redux";
import { setRegister, postRegister } from "../../modules/main";

const RegisterPageContainer = connect(
  (state) => ({
    state: state.register,
  }),
  (dispatch) => ({
    setRegister: (data) => dispatch(setRegister(data)),
    postRegister: (data) => dispatch(postRegister(data)),
  })
)(RegisterPage);

export default RegisterPageContainer;
