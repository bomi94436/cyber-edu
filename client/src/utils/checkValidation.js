import * as data from "./sampleData";

const studentIdRegExp = /^\d+$/; // 숫자만 입력
const passwordRegExp = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/; // 영어, 숫자 조합해서 8자리 이상 입력
const emailRegExp = /^[\w-]+(\.[\w-]+)*@([a-z0-9-]+(\.[a-z0-9-]+)*?\.[a-z]{2,6}|(\d{1,3}\.){3}\d{1,3})(:\d{4})?$/;
const phoneNumberRegExp = /^\d{3}-\d{3,4}-\d{4}$/;

export const validate = (target, value, password) => {
  switch (target) {
    case "role":
      if (value) return true;
      else return false;

    case "studentId":
      if (value.length === 9 && value.match(studentIdRegExp)) return true;
      else return false;

    case "password":
      if (
        value.length >= 8 &&
        value.length <= 20 &&
        value.match(passwordRegExp)
      )
        return true;
      else return false;

    case "re_password":
      if (value === password) return true;
      else return false;

    case "name":
      if (value.length >= 2) return true;
      else return false;

    case "email":
      if (value.match(emailRegExp)) return true;
      else return false;

    case "phone":
      if (value.match(phoneNumberRegExp)) return true;
      else return false;

    default:
      return null;
  }
};

export const inputClassNameByValid = (valid) => {
  switch (valid) {
    case true:
      return "is-valid";

    case false:
      return "is-invalid";

    default:
      return "";
  }
};

export const isEveryFieldValid = (state) => {
  for (const [key, value] of Object.entries(state)) {
    if (!value) return false;
  }
  return true;
};
