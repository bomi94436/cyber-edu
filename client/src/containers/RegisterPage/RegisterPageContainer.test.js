import React from "react";
import nock from "nock";
import { mount } from "enzyme";
import { Provider } from "react-redux";

import RegisterPageContainer from "./RegisterPageContainer";
import store from "../../modules/store";
import * as data from "../../utils/sampleData";

describe("RegisterPageContainer", () => {
  let component = null;

  it("renders properly", () => {
    component = mount(
      <Provider store={store}>
        <RegisterPageContainer />
      </Provider>
    );
  });

  it("matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("dispatches SET_REGISTER action", () => {
    const value = data.register[1].value;
    const mockedEvent = {
      target: { value: value },
    };
    component.find("#studentId-input").simulate("change", mockedEvent);
    expect(store.getState().register.studentId).toBe(value);
  });

  it("fetches and updates using postRegister", async () => {
    const dataToSubmit = {};
    data.register.forEach((element) => {
      if (element.name === "studentId" || element.name === "phone") {
        dataToSubmit[`${element.name}`] = Number(element.value);
      } else {
        dataToSubmit[`${element.name}`] = element.value;
      }
    });
    nock("http://localhost")
      .post("/api/auth/register", dataToSubmit)
      .once()
      .reply(200, {
        message: "회원가입 완료",
      });
    component.find("form").simulate("submit");

    const waitForNextAction = new Promise((resolve) => {
      const unsubscribe = store.subscribe(() => {
        resolve();
        unsubscribe();
      });
    });
    await waitForNextAction;
    expect(component.find("#studentId-input").props().value).toBe(undefined);
  });
});
