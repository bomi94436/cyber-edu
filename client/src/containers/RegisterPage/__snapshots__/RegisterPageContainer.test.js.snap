// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPageContainer matches snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(RegisterPage)>
    <RegisterPage
      postRegister={[Function]}
      setRegister={[Function]}
      state={
        Object {
          "valid": Object {
            "email": null,
            "name": null,
            "password": null,
            "phone": null,
            "re_password": null,
            "role": null,
            "studentId": null,
          },
          "value": Object {
            "email": "",
            "name": "",
            "password": "",
            "phone": "",
            "re_password": "",
            "role": "",
            "studentId": "",
          },
        }
      }
    >
      <div
        className="register"
      >
        <form
          className="form-frame"
          onSubmit={[Function]}
        >
          <RegisterRadio
            name="role"
            setRegister={[Function]}
            title="학생"
            value="student"
          >
            <span>
              <label
                htmlFor="student-input"
              >
                학생
              </label>
              <input
                id="student-input"
                name="role"
                onChange={[Function]}
                type="radio"
                value="student"
              />
            </span>
          </RegisterRadio>
          <RegisterRadio
            name="role"
            setRegister={[Function]}
            title="교수"
            value="professor"
          >
            <span>
              <label
                htmlFor="professor-input"
              >
                교수
              </label>
              <input
                id="professor-input"
                name="role"
                onChange={[Function]}
                type="radio"
                value="professor"
              />
            </span>
          </RegisterRadio>
          <RegisterRow
            dispatchName="studentId"
            placeholder="202012345"
            setRegister={[Function]}
            title="학번(직번)"
            type="text"
            valid={null}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="studentId-input"
              >
                학번(직번)
              </label>
              <input
                className="form-control "
                id="studentId-input"
                onChange={[Function]}
                placeholder="202012345"
                type="text"
              />
              <br />
            </div>
          </RegisterRow>
          <RegisterRow
            dispatchName="password"
            placeholder="**********"
            setRegister={[Function]}
            title="비밀번호"
            type="password"
            valid={null}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="password-input"
              >
                비밀번호
              </label>
              <input
                className="form-control "
                id="password-input"
                onChange={[Function]}
                placeholder="**********"
                type="password"
              />
              <br />
            </div>
          </RegisterRow>
          <RegisterRow
            dispatchName="re_password"
            placeholder="**********"
            setRegister={[Function]}
            title="비밀번호 확인"
            type="password"
            valid={null}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="re_password-input"
              >
                비밀번호 확인
              </label>
              <input
                className="form-control "
                id="re_password-input"
                onChange={[Function]}
                placeholder="**********"
                type="password"
              />
              <br />
            </div>
          </RegisterRow>
          <RegisterRow
            dispatchName="name"
            placeholder="홍길동"
            setRegister={[Function]}
            title="이름"
            type="text"
            valid={null}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="name-input"
              >
                이름
              </label>
              <input
                className="form-control "
                id="name-input"
                onChange={[Function]}
                placeholder="홍길동"
                type="text"
              />
              <br />
            </div>
          </RegisterRow>
          <RegisterRow
            dispatchName="email"
            placeholder="test@example.com"
            setRegister={[Function]}
            title="이메일"
            type="email"
            valid={null}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="email-input"
              >
                이메일
              </label>
              <input
                className="form-control "
                id="email-input"
                onChange={[Function]}
                placeholder="test@example.com"
                type="email"
              />
              <br />
            </div>
          </RegisterRow>
          <RegisterRow
            dispatchName="phone"
            placeholder="010-1234-5678"
            setRegister={[Function]}
            title="휴대폰 번호"
            type="text"
            valid={null}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="phone-input"
              >
                휴대폰 번호
              </label>
              <input
                className="form-control "
                id="phone-input"
                onChange={[Function]}
                placeholder="010-1234-5678"
                type="text"
              />
              <br />
            </div>
          </RegisterRow>
          <button
            className="btn btn-primary btn-block"
            disabled={true}
            type="submit"
          >
            회원가입
          </button>
        </form>
      </div>
    </RegisterPage>
  </Connect(RegisterPage)>
</Provider>
`;
